x-pack/plugins/alerting/server/saved_objects/migrations.ts

```diff
@@ -333,6 +342,64 @@ function restructureConnectorsThatSupportIncident(
   };
 }
 
+function removeNullsFromSecurityRules(
+  doc: SavedObjectUnsanitizedDoc<RawAlert>
+): SavedObjectUnsanitizedDoc<RawAlert> {
+  const {
+    attributes: { params },
+  } = doc;
+  return {
+    ...doc,
+    attributes: {
+      ...doc.attributes,
+      params: {
+        ...params,
+        buildingBlockType: params.buildingBlockType != null ? params.buildingBlockType : undefined,
+        note: params.note != null ? params.note : undefined,
```

Comment:
nit: there's so many of these, it might be useful (if possible, with the typing and all) to create a function to do this, so you can just use

```
   note: convertNullToUndefined(params.note)
```